cmd/jaeger/internal/integration/storagecleaner/extension.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package storagecleaner
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/extension"
+
+	"github.com/jaegertracing/jaeger/cmd/jaeger/internal/extension/jaegerstorage"
+	"github.com/jaegertracing/jaeger/storage"
+)
+
+var (
+	_ extension.Extension = (*storageCleaner)(nil)
+	_ extension.Dependent = (*storageCleaner)(nil)
+)
+
+const (
+	Port = "9231"
+	URL  = "/purge"
+)
+
+type storageCleaner struct {
+	config *Config
+	server *http.Server
+}
+
+func newStorageCleaner(config *Config) *storageCleaner {
+	return &storageCleaner{
+		config: config,
+	}
+}
+
+func (c *storageCleaner) Start(ctx context.Context, host component.Host) error {
+	storageFactory, err := jaegerstorage.GetStorageFactory(c.config.TraceStorage, host)
+	if err != nil {
+		return fmt.Errorf("cannot find storage factory for Badger: %w", err)
+	}
+
+	purgeStorage := func() error {
+		purger := storageFactory.(storage.Purger)
```

Comment:
we should not assume every factory implements this.
```suggestion
		purger, ok := storageFactory.(storage.Purger)
		if !ok {
		  return fmt.Errorf("storage %s does not implement Purger interface", c.config.TraceStorage)
		}
```